interfaces/builtin/login_session_control.go

```diff
@@ -0,0 +1,69 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const loginSessionControlSummary = `allows setup of login session & seat`
+
+const loginSessionControlBaseDeclarationSlots = `
+  login-session-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const loginSessionControlConnectedPlugAppArmor = `
+# Description: Can setup login session & seat.
+
+#include <abstractions/dbus-strict>
+
+dbus (send,receive)
+    bus=system
+    path=/org/freedesktop/login1/{seat,session}/*
+    interface=org.freedesktop.DBus.Properties
+    member={GetAll,PropertiesChanged,Get}
+    peer=(label=unconfined),
+
+dbus (send,receive)
+    bus=system
+    path=/org/freedesktop/login1/seat/*
+    interface=org.freedesktop.login1.Seat
+    member={ActiveSession,SwitchTo}
+    peer=(label=unconfined),
+
+dbus (send,receive)
+    bus=system
+    path=/org/freedesktop/login1/session/*
+    interface=org.freedesktop.login1.Session
+    member={TakeControl,TakeDevice,PauseDevice,PauseDeviceComplete,ResumeDevice,ReleaseDevice,Active,State,Lock,Unlock}
+    peer=(label=unconfined),
+`
```

Comment:
Should this rule also be present:
```
/{,usr/}{,s}bin/loginctl ixr,
```

Importantly, I do not want to add this rule unless it is known to work with only this interface connected since systemd commands have a tendency to spin out wrt to system access. Eg, systemctl ends up requiring the equivalent of device ownership, but hostnamectl works very cleanly with confinement. This could be demonstrated via spread tests.

Reply:
Empirically, adding this rule doesn't break anything, but I don't know enough to know whether this should be present.

Reply:
We don't need this, we're not using `loginctl` - just talking to logind via dbus.